sdk/azidentity/azure_cli_credential_client.go

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azidentity
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os/exec"
+	"regexp"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+)
+
+const (
+	DefaultSuffix        = "/.default"
+	AzureCLINotInstalled = "Azure CLI not installed"
+	AzNotLogIn           = "ERROR: Please run 'az login'"
+	WinAzureCLIError     = "'az' is not recognized"
+	invalidResourceError = "Resource is not in expected format. Only alphanumeric characters, '.', ':', '-', and '/' are allowed"
+)
+
+type azureCLIAccessToken struct {
+	Token        string `json:"accessToken"`
```

Comment:
Let's make the fields private too so that readers of the code know that the fields are intended not to be public.